### PR TITLE
Add frontend lint / tests to ci workflow

### PR DESCRIPTION
**Pull request by acrantel**
makes frontend2 tests / linting run on each pr.

### CODE REVIEW

**Review by lowtorola — APPROVED** [review]

**Review by n8kim1 — DISMISSED** [review]
so nice to have

**Comment by n8kim1** [comment]
do we ever plan on renaming frontend2 to frontend and then frontend to "old" or smth? if so, we'll have to make careful note to change this folder too, cuz it's not really a directory name in the same way in code as everything else

to be fair, i think that errors will expose this too

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,6 +17,39 @@ env:
...
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Install package dependencies
+        run: cd frontend2 && npm install
+      - name: Check linting and formatting
+        # Custom script for checking the linting and formatting being in place
+        run: cd frontend2 && npm run lint
```
Comment (j-mao) [inline .github/workflows/ci.yml:49]: See the `working-directory` key in the example [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun).

**Comment by j-mao** [comment]
GitHub workflow looks good to me! [modulo nit about using `working-directory` instead of `cd blah &&`] Thanks so much for doing this :tada:



> do we ever plan on renaming frontend2 to frontend and then frontend to "old" or smth? if so, we'll have to make careful note to change this folder too, cuz it's not really a directory name in the same way in code as everything else

This can be done already, can't it? If I understand correctly this branch has zero dependency on the contents of `frontend` so the old stuff can be safely dropped and this directory renamed on the branch. Whenever you need to peek at old stuff, you can still find it in version control on `main` until this big feature branch gets merged.

**Comment by acrantel** [comment]
> GitHub workflow looks good to me! [modulo nit about using `working-directory` instead of `cd blah &&`] Thanks so much for doing this 🎉
> 
> > do we ever plan on renaming frontend2 to frontend and then frontend to "old" or smth? if so, we'll have to make careful note to change this folder too, cuz it's not really a directory name in the same way in code as everything else
> 
> This can be done already, can't it? If I understand correctly this branch has zero dependency on the contents of `frontend` so the old stuff can be safely dropped and this directory renamed on the branch. Whenever you need to peek at old stuff, you can still find it in version control on `main` until this big feature branch gets merged.

will put this into our roadmap

**Comment by acrantel** [comment]
added a recommended plugin to eslint and required semicolons.

**Review by n8kim1 — APPROVED** [review]

**Pull request merged**